lib/cfn-nag/custom_rules/CognitoUserPoolMfaConfigurationOnorOptionalRule.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require_relative 'base'
+
+class CognitoUserPoolMfaConfigurationOnorOptionalRule < BaseRule
+  def rule_text
+    "AWS Cognito UserPool should have MfaConfiguration set to 'ON' (MUST be wrapped in quotes) or at least 'OPTIONAL'"
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F85'
+  end
+
+  def audit_impl(cfn_model)
+    violating_userpools = cfn_model.resources_by_type('AWS::Cognito::UserPool').select do |userpool|
+      violating_userpool?(userpool)
+    end
+
+    violating_userpools.map(&:logical_resource_id)
+  end
+
+  private
+
+  def yaml_reserved_keyword_boolean_conv?(user_pool, param_value = '')
+    if [true, false].include?(user_pool.mfaConfiguration)
```

Comment:
there is a truthy? method in utils somewhere that you should likely use instead

Reply:
Made an update to use the truthy? and not_truthy? methods.